src/clj_puppetdb/core.clj

```diff
@@ -18,29 +19,59 @@
   ([^String host] (http/make-http-client host))
   ([^String host opts] (http/make-https-client host opts)))
 
+(defn query-with-metadata
+  "Use the given PuppetDB client to query the server.
+
+  The path argument should be a valid endpoint, e.g. \"/v4/nodes\".
+
+  The query-vec argument is optional, and should be a vector representing an API query,
+  e.g. [:= [:fact \"operatingsystem\"] \"Linux\"]
+
+  The params map is optional, and can contain any of the following keys:
+  - :order-by (a vector of maps, each specifying a :field and an :order key)
+  - :limit (the number of results to request)
+  - :offset (the index of the first result to return, defaults to 0)
+  - :include-total (boolean indicating whether to return the total number of records available)
+  For example: `{:limit 100 :offset 0 :order-by [{:field \"value\" :order \"asc\"}] :include-total true}`
+
+  This function returns two maps in a vector. The first map is the query result as returned by 'query'. The second
+  contains additional metadata. Currently the only supported kind of metadata is:
+  - :total (the total number of records available)"
+  ([client path params]
```

Comment:
Could possibly use named rest parameters here to prevent all of the nil parameter passing?